### ISSUE TITLE
`import Distributed; Distributed.@everywhere x = 2` does not work

### ISSUE THREAD

**Issue by cossio** [JuliaLang/julia#28966]
This works fine:

```julia
using Distributed; @everywhere x = 2
```

but

```julia
import Distributed; Distributed.@everywhere x = 2
# ERROR: LoadError: UndefVarError: @everywhere not defined
```

gives an error.

**Comment by vchuravy** [comment]
```
julia> import Distributed; Distributed.@everywhere x = 2
ERROR: LoadError: UndefVarError: @everywhere not defined
in expression starting at C:\cygwin\home\Administrator\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.0\Distributed\src\macros.jl:182
```

Looks like the macro https://github.com/JuliaLang/julia/blob/2346be896129f594ce2e6104f612828468e63bfc/stdlib/Distributed/src/macros.jl#L182 assumes that `@everywhere` is in the namespace. 

d61b054118dd78ab2360cea310a0e0c42f710153 and x-ref 23164

**Comment by ibadr** [comment]
Also, this fails with the same error, despite ```using Distributed```

```julia
const parallel = true
if !parallel
    println("not parallel")
else
    using Distributed; @everywhere x = 2
    println("parallel")
end
```

**Comment by ibadr** [comment]
For now, this slight tweak works for me
```julia
const parallel = true
if !parallel
    println("not parallel")
else
    using Distributed
    Distributed.remotecall_eval(Main, procs(), :(x = 2)) # in lieu of @everywhere x = 2
    println("parallel")
end
```

**Comment by martinholters** [comment]
@ibadr That's a different issue: The parser tries to expand the macro before the `using` is evaluated, because the parser processes the whole `if`/`else`/`end` before anything of it is evaluated.

**Comment by fredrikekre** [comment]
GitHub doesn't autoclose issues from commit messages any more? fixed by #32144
